### ISSUE TITLE
Game starts in non-primary language on OSX

### ISSUE THREAD

**Issue by nickretallack** [CleverRaven/Cataclysm-DDA#20665]
I'm running Mac OS X 10.12.3 (Sierra).  Under System Preferences > Languages & Region, my Preferred Languages list looks like this:

English (U.S) -- Primary
Japanese

I downloaded http://dev.narc.ro/cataclysm/jenkins-latest/OSX/Tiles/Cataclysm-0.C-6356.dmg and ran it, and the interface was entirely in Japanese.

**Comment by Leland** [comment]
Can confirm

**Comment by OzoneH3** [comment]
I've setup a osx 10.13 vm to try this but everything is fine here.

@Leland Can you post a screenshot of your languages window and the output of 'locale'

**Comment by Leland** [comment]
@OzoneH3 I'll give it a go again soon, do note my comment over yonder as well: https://github.com/CleverRaven/Cataclysm-DDA/issues/20713#issuecomment-324495980

**Comment by sxmichaels** [comment]
Aha, I think I've figured out what causes this as well as #22355. It turns out that OSX itself doesn't (consistently?) set the locale environment variables at all… if I run `locale`, for instance, I get the settings below — which is probably what happens to other OSX users unless they've manually reset one of these variables in their shell configs.

```
LANG=
LC_COLLATE="C"
LC_CTYPE="UTF-8"
LC_MESSAGES="C"
LC_MONETARY="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_ALL=
```
Because of this, the invocation to `setlocale(LC_ALL, "")` at [`main.cpp:410`](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/main.cpp#L409) sets the program locale _back_ to `C`. This is what causes CJK text to turn to garbage, and why setting `LC_ALL` manually worked for me.

Subsequently, `set_language()` doesn't set the `LANGUAGE` env var if `System Language` is set, but OSX doesn't set `LANGUAGE` automatically, so it defaults to English.

It looks like the correct way to get the system language on OSX is to use the [CFLocale](https://developer.apple.com/documentation/corefoundation/cflocale-rsj) library, which fortunately can be called from C++. I will work on a PR tonight to incorporate both of these fixes by using CFLocale — I just need to figure out how to get the language shortcodes out of it.

**Comment by OzoneH3** [comment]
Fixed with #22513
